tests/impl/repository/elasticsearch_repo/elements.py

```diff
@@ -0,0 +1,51 @@
+# Standard Library Imports
+import re
+
+from datetime import datetime
+
+# Protean
+from protean.core.aggregate import BaseAggregate
+from protean.core.exceptions import ValidationError
+from protean.core.field.basic import DateTime, Integer, String
+from protean.core.field.embedded import ValueObjectField
+from protean.core.repository.base import BaseRepository
+from protean.core.value_object import BaseValueObject
+
+
+class Person(BaseAggregate):
+    first_name = String(max_length=50, required=True)
```

Comment:
The mapping for this would be 
```
{
    "mappings": {
        "first_name": {
            "type": "text",
            "fields": {
                "keyword": {
                    "type": "keyword"
                }
            }
        }
    }
}
```

Reply:
Yes, that's correct. That's the mapping generated by default.

Reply:
This is what is generated for the aggregate by default:

```
{
  "person" : {
    "aliases" : { },
    "mappings" : {
      "properties" : {
        "age" : {
          "type" : "long"
        },
        "created_at" : {
          "type" : "date"
        },
        "first_name" : {
          "type" : "text",
          "fields" : {
            "keyword" : {
              "type" : "keyword",
              "ignore_above" : 256
            }
          }
        },
        "last_name" : {
          "type" : "text",
          "fields" : {
            "keyword" : {
              "type" : "keyword",
              "ignore_above" : 256
            }
          }
        }
      }
    },
    "settings" : {
      "index" : {
        "creation_date" : "1569472187099",
        "number_of_shards" : "1",
        "number_of_replicas" : "1",
        "uuid" : "lBOf5_ZdTxugXcOyj4xeXQ",
        "version" : {
          "created" : "7030299"
        },
        "provided_name" : "person"
      }
    }
  }
}
```